appengine/datastore/src/test/java/com/example/appengine/QueriesTest.java

```diff
@@ -770,17 +770,14 @@ public void queryRestrictions_surprisingMultipleValuesTwoNotEquals_returnsMatche
     // [START surprising_behavior_example_4]
     Query q =
         new Query("Widget")
-            .setFilter(new FilterPredicate("x", FilterOperator.NOT_EQUAL, 1))
-            .setFilter(new FilterPredicate("x", FilterOperator.NOT_EQUAL, 2));
+            .setFilter(
+                CompositeFilterOperator.and(
```

Comment:
I added a link to the docs.
https://cloud.google.com/appengine/docs/java/datastore/query-restrictions#properties_with_multiple_values_can_behave_in_surprising_ways

That link describes that the query acts like
```
x < 1 OR (x > 1 AND x < 2) OR x > 2
```